marshmallow_sqlalchemy/schema.py

```diff
@@ -200,3 +225,24 @@ def validate(self, data, session=None, *args, **kwargs):
         if not self.session:
             raise ValueError('Validation requires a session')
         return super(ModelSchema, self).validate(data, *args, **kwargs)
+
+    def _split_model_kwargs_association(self, data):
+        """Split serialized attrs to ensure association proxies are passed separately.
+
+        This is necessary for Python < 3.6.0, as the order in which kwargs are passed
+        is non-deterministic, and associations must be parsed by sqlalchemy after their
+        intermediate relationship, unless their `creator` has been set.
+
+        :param data: serialized dictionary of attrs to split on association_proxy.
+        """
+        association_attrs = {
+            key: value
+            for key, value in iteritems(data)
+            if isinstance(getattr(self.opts.model, key), AssociationProxy)
```

Comment:
This will fail when `unknown = INCLUDE` is set on the Schema.

The following test errors:

```python
    def test_transient_load_with_unknown_include(self, models, session, school):
        class SchoolSchemaTransient(ModelSchema):
            class Meta:
                model = models.School
                sqla_session = session
                unknown = marshmallow.INCLUDE

        sch = SchoolSchemaTransient()
        dump_data = unpack(sch.dump(school))
        dump_data['foo'] = 'bar'
        load_data = unpack(sch.load(dump_data, transient=True))
        assert isinstance(load_data, models.School)
        state = sa.inspect(load_data)
        assert state.transient
```

```
AttributeError: type object 'School' has no attribute 'foo'
```

Reply:
Hmm very good point, updated the model kwarg extractor to ignore unknown keys. This test is added and passing